torch/_dynamo/decorators.py

```diff
@@ -437,19 +437,28 @@ class directly; instead, use :func:`mark_dynamic`.
 
 
 @forbid_in_graph
-def mark_unbacked(t, index):
+def mark_unbacked(t, index, strict=False):
     """
     Mark a tensor as having an unbacked dim.  This changes the semantics of operations,
     we will always report the size does not equal zero/one, we will turn asserts
     on this index into runtime asserts, and if you try to get the real value we will
     raise an exception.  In other words, we will treat this dimension as if it was
     data dependent (we do not know anything about its value.)
+
+    For historical reasons, by default if an unbacked dim is specialized, we will
```

Comment:
how much work does it take to break the history, i think alternativly we can make this consistent with mark_dynamic by making strict=False default for mark dynamic?

Reply:
I think in general breaking backwards compatibility is a big deal. We could explore doing it internally via a just knob though.